adjust_test.go

```diff
@@ -502,4 +559,194 @@ func TestAdjustFormula(t *testing.T) {
+	f := NewFile()
+	assert.NoError(t, f.SetCellFormula("Sheet1", "B1", "=A1+A2"))
+	assert.NoError(t, f.InsertRows("Sheet1", 2, 1))
+	formula, err := f.GetCellFormula("Sheet1", "B1")
+	assert.NoError(t, err)
+	assert.Equal(t, "A1+A3", formula)
+	assert.NoError(t, f.RemoveRow("Sheet1", 2))
+	formula, err = f.GetCellFormula("Sheet1", "B1")
+	assert.NoError(t, err)
+	assert.Equal(t, "A1+A2", formula)
+
+	// Test insert columns in middle of range (+ notation)
+	f = NewFile()
+	assert.NoError(t, f.SetCellFormula("Sheet1", "A1", "=B1+C1"))
+	assert.NoError(t, f.InsertCols("Sheet1", "C", 1))
+	formula, err = f.GetCellFormula("Sheet1", "A1")
+	assert.NoError(t, err)
+	assert.Equal(t, "B1+D1", formula)
+	assert.NoError(t, f.RemoveCol("Sheet1", "C"))
+	formula, err = f.GetCellFormula("Sheet1", "A1")
+	assert.NoError(t, err)
+	assert.Equal(t, "B1+C1", formula)
+
+	// Test insert row and columns in a rectangular range (+ notation)
+	f = NewFile()
+	assert.NoError(t, f.SetCellFormula("Sheet1", "A1", "=D4+D5+E4+E5"))
+	assert.NoError(t, f.InsertCols("Sheet1", "E", 1))
+	assert.NoError(t, f.InsertRows("Sheet1", 5, 1))
+	formula, err = f.GetCellFormula("Sheet1", "A1")
+	assert.NoError(t, err)
+	assert.Equal(t, "D4+D6+F4+F6", formula)
+
+	// Test insert row in middle of range (: notation)
+	f = NewFile()
+	formulaType, reference := STCellFormulaTypeArray, "B1:B1"
+	assert.NoError(t, f.SetCellFormula("Sheet1", "B1", "=A1:A2", FormulaOpts{Ref: &reference, Type: &formulaType}))
+	assert.NoError(t, f.InsertRows("Sheet1", 2, 1))
+	formula, err = f.GetCellFormula("Sheet1", "B1")
+	assert.NoError(t, err)
+	assert.Equal(t, "A1:A3", formula)
+	assert.NoError(t, f.RemoveRow("Sheet1", 2))
+	formula, err = f.GetCellFormula("Sheet1", "B1")
+	assert.NoError(t, err)
+	assert.Equal(t, "A1:A2", formula)
+
+	// Test insert columns in middle of range (: notation)
+	f = NewFile()
+	formulaType, reference = STCellFormulaTypeArray, "A1:A1"
+	assert.NoError(t, f.SetCellFormula("Sheet1", "A1", "=B1:C1", FormulaOpts{Ref: &reference, Type: &formulaType}))
+	assert.NoError(t, f.InsertCols("Sheet1", "C", 1))
+	formula, err = f.GetCellFormula("Sheet1", "A1")
+	assert.NoError(t, err)
+	assert.Equal(t, "B1:D1", formula)
+	assert.NoError(t, f.RemoveCol("Sheet1", "C"))
+	formula, err = f.GetCellFormula("Sheet1", "A1")
+	assert.NoError(t, err)
+	assert.Equal(t, "B1:C1", formula)
+
+	// Test insert row and columns in a rectangular range (: notation)
+	f = NewFile()
+	formulaType, reference = STCellFormulaTypeArray, "A1:A1"
+	assert.NoError(t, f.SetCellFormula("Sheet1", "A1", "=D4:E5", FormulaOpts{Ref: &reference, Type: &formulaType}))
+	assert.NoError(t, f.InsertCols("Sheet1", "E", 1))
+	assert.NoError(t, f.InsertRows("Sheet1", 5, 1))
+	formula, err = f.GetCellFormula("Sheet1", "A1")
+	assert.NoError(t, err)
+	assert.Equal(t, "D4:F6", formula)
+}
+
+func TestCellNotOffsetButFormulaAffected(t *testing.T) {
+	f := NewFile()
+	assert.NoError(t, f.SetCellFormula("Sheet1", "B1", "=A3+A4"))
```

Comment:
The equal symbol in the formula functions can be ignored when set cell formula in all test cases.